spring-boot-project/spring-boot-tools/spring-boot-maven-plugin/src/main/java/org/springframework/boot/maven/ArtifactsLibraries.java

```diff
@@ -164,12 +164,15 @@ private boolean isLocal(Artifact artifact) {
 
 	private String getFileName(Artifact artifact) {
 		StringBuilder sb = new StringBuilder();
-		sb.append(artifact.getArtifactId()).append("-").append(artifact.getBaseVersion());
+		sb.append(artifact.getArtifactId()).append("-").append(artifact.getGroupId()).append("-")
```

Comment:
Rather than doing this here, I think it would be better to reuse [the existing logic that prepends the group ID when a duplicate is detected](https://github.com/spring-projects/spring-boot/blob/a1c8997f615a66da7cd471f87921830ea0408ba7/spring-boot-project/spring-boot-tools/spring-boot-maven-plugin/src/main/java/org/springframework/boot/maven/ArtifactsLibraries.java#L114-L118). A configuration option should be added so that, when enabled, the group ID always prepended irrespective of whether or not a duplicate has been detected.

Reply:
Hi @wilkinsona ,
method `` doWithLibraries`` it is using  method `getDuplicates` and it's using `getFileName` and on loop for every file \ Artifacts we are using that method this is why I thought we can fix only at method `getFileName`
how can I inject that opinion ?

Reply:
Hi @wilkinsona ,
do you think that ```getDuplicates``` should return Artifices and then it can be better compare ?

Reply:
I don't think there's anything to compare. The file name already contains the artifact ID, version, and any classifier. 
Maven guarantees that within a group, each combination of artifact ID, version, and classifier will be unique. Therefore, prepending the group ID is sufficient to avoid a duplicate entry name.